### DISCUSSION TITLE
Release Maintenance Strategy

### DISCUSSION THREAD

**Discussion by harrylewis** [lostisland/faraday#1413] [Help] [answered]
Hey there 👋🏻 

I'm curious about the release maintenance strategy that is used here at Faraday. For instance, I see new versions on the `0.x` release being made in 2022, alongside new versions for `1.x` and `2.x`. What kind of updates is the `0.x` line receiving at this time?

Additionally, does Faraday aim to follow [SemVer](https://semver.org/) with these version updates?

Thanks in advance for the insight!

**Comment by iMacTia** [comment, marked as answer]
Hi @harrylewis we're already following SemVer for our releases 🙌.
The reason for 0.x and 1.x versions being pushed is that they're still widely being used out in the wild.

We obviously recommend people to use Faraday 2.x, but since Faraday is what is colloquially called a "plumbing" gem (i.e. used as a dependency of other gems) people often don't have direct control over which version of Faraday they can use.
Updates to the 0.x and 1.x versions are mostly for fixes and ease of upgrades, there are no plans to push new features over these as well.

I hope this helps clarifying, but if you have any other follow-up question I'd be happy to help!

**Reply by iMacTia** [reply]
P.S.: I find GitHub default sorting of releases quite confusing, I wish they'd allow you to sort by version rather than release date 😕

**Reply by harrylewis** [reply]
Thank you for such a quick reply @iMacTia! This all makes sense to me.
